src/Paket.Core/Dependencies/PackageResolver.fs

```diff
@@ -649,14 +652,56 @@ let Resolve (getVersionsF, getPackageDetailsF, groupName:GroupName, globalStrate
 
     let rec step (stage:Stage) (stackpack:StackPack) compatibleVersions (flags:StepFlags) =
 
-        let inline fuseConflicts currentConflict priorConflictSteps =
-            match currentConflict, priorConflictSteps with
-            | currentConflict, (lastConflict,lastStep,lastRequirement,lastCompatibleVersions,lastFlags)::priorConflictSteps -> 
-                let continueConflict = 
-                    { currentConflict with VersionsToExplore = lastConflict.VersionsToExplore }        
-                step (Inner((continueConflict,lastStep,lastRequirement),priorConflictSteps))  stackpack lastCompatibleVersions lastFlags 
-            | currentConflict, [] -> currentConflict
-
+      //printfn "--step-- %A-%A-%A-%A" stage stackpack compatibleVersions flags
+
+        let inline fuseConflicts currentConflict priorConflictSteps conflicts =
+                    let findMatchingStep (currentConflict:ConflictState) priorConflictSteps =
+                        let row =
+                            priorConflictSteps
+                            |> Seq.tryFind (fun (lastConflict:ConflictState,lastStep:ResolverStep,lastRequirement:PackageRequirement,lastCompatibleVersions:seq<SemVerInfo*_>,lastFlags:StepFlags) ->
+                                let currentNames =
+                                    conflicts
+                                    |> Seq.collect (fun c ->
+                                        let graphNameList =
+                                            c.Graph
+                                            |> List.map (fun (pr:PackageRequirement) -> pr.Name) 
+                                        c.Name :: graphNameList)
+                                    |> Seq.toArray
+                                currentNames |> Array.contains lastRequirement.Name                        
+                            )
+                        row
+                        |> Option.map (fun r -> r, priorConflictSteps |> List.filter(fun r2 -> not (obj.ReferenceEquals(r, r2))))
+                    match priorConflictSteps with
+                    | [] -> currentConflict
+                    | priorConflictSteps ->
+                        match findMatchingStep currentConflict priorConflictSteps with
+                        | Some((lastConflict,lastStep,lastRequirement,lastCompatibleVersions,lastFlags), priorConflictSteps) ->
+                            let continueConflict = 
+                                { currentConflict with VersionsToExplore = lastConflict.VersionsToExplore }        
+                            step (Inner((continueConflict,lastStep,lastRequirement),priorConflictSteps))  stackpack lastCompatibleVersions lastFlags
+                        | None ->
+                            // could not find a specific package - go back one
+                            let (lastConflict,lastStep,lastRequirement,lastCompatibleVersions,lastFlags) :: priorConflictSteps = priorConflictSteps
```

Comment:
Is this supposed to pull the head off of `priorConflictSteps` as `(lastConflict,lastStep,lastRequirement,lastCompatibleVersions,lastFlags)` and then alias it's tail as `priorConflictSteps` ?